nixos/modules/services/misc/jellyfin.nix

```diff
@@ -35,7 +35,26 @@ in
         Group = cfg.group;
         StateDirectory = "jellyfin";
         CacheDirectory = "jellyfin";
-        ExecStart = "${pkgs.jellyfin}/bin/jellyfin --datadir '/var/lib/${StateDirectory}' --cachedir '/var/cache/${CacheDirectory}'";
+        db_version = pkgs.lib.versions.majorMinor pkgs.jellyfin.version;
+        ExecPreStart = ''
+          #!/bin/bash
+          mkdir -p /var/lib/${StateDirectory}/database/${db_version}
+          if [ "${db_version}" = "10.6.1" ]; then
+            cp -r /var/lib/${StateDirectory}/data /var/lib/${StateDirectory}/database/${db_version}/
+            exit 0
+          else
+            versions=$(find /var/lib/${StateDirectory}/database/ -maxdepth 1 -mindepth 1 | sort -r)
+            for version in versions;
+            do
+              if [ -d "$version" ]; then
+                if find "$version" -mindepth 1 -print -quit 2>/dev/null | grep -q .; then;
+                  cp -r "$version" /var/lib/${StateDirectory}/database/${db_version}
```

Comment:
If we have this sort of thing, I guess we might want a clean-up mechanism (or not, I really do not know). Maybe drop all but 2 most recent?